test_genie_python_dae.py

```diff
@@ -82,6 +83,90 @@ def test_GIVEN_running_instrument_WHEN_pars_changed_THEN_pars_saved_in_file(self
         else:
             self.assertEqual(0, saved_beamstop)
 
+    def test_GIVEN_wait_for_complete_callback_dae_settings_is_false_and_valid_tables_given_THEN_dae_does_not_wait_and_xml_values_are_not_initially_correct(self):
+
+        set_wait_for_complete_callback_dae_settings(False)
+        set_genie_python_raises_exceptions(True)
+
+        table_path_template = r"{}\tables\{}"
+        wiring = table_path_template.format(os.environ["ICPCONFIGROOT"], "f_wiring_doors_all_event_process_5.dat")
+        detector = table_path_template.format(os.environ["ICPCONFIGROOT"], "det_corr_184_process_5.dat")
+        spectra = table_path_template.format(os.environ["ICPCONFIGROOT"], "f_spectra_doors_all_process_2to1_5.dat")
+
+        with self.assertRaises(ValueError):
+            g.change_tables(wiring, detector, spectra)
+
+        set_genie_python_raises_exceptions(False)
+
+    def test_GIVEN_wait_for_complete_callback_dae_settings_is_true_and_valid_tables_given_THEN_dae_waits_and_xml_values_are_confirmed_correct(self):
+
+        set_wait_for_complete_callback_dae_settings(True)
+        g.change_tcb(0, 10000, 100, regime=2)
+
+        table_path_template = r"{}\tables\{}"
+        wiring = table_path_template.format(os.environ["ICPCONFIGROOT"], "f_wiring_doors_all_event_process_5.dat")
+        detector = table_path_template.format(os.environ["ICPCONFIGROOT"], "det_corr_184_process_5.dat")
+        spectra = table_path_template.format(os.environ["ICPCONFIGROOT"], "f_spectra_doors_all_process_2to1_5.dat")
+        g.change_tables(wiring, detector, spectra)
+
+    def test_GIVEN_valid_spectra_table_to_change_tables_THEN_get_spectra_table_returns_correct_file_path(self):
+
+        set_wait_for_complete_callback_dae_settings(True)
+        g.change_tcb(0, 10000, 100, regime=2)
+        #forward slashes are required for comparison here as ISIS ICP returns forward slashes
+        spectra = r"{}/tables/RCPTT_{}128.dat".format(os.environ["ICPCONFIGROOT"], "spectra")
+
+        g.change_tables(spectra=spectra)
+        self.assertEqual(g.get_spectra_table(), spectra)
+
+    def test_GIVEN_valid_wiring_table_to_change_tables_THEN_get_wiring_table_returns_correct_file_path(self):
+
+        set_wait_for_complete_callback_dae_settings(True)
+        g.change_tcb(0, 10000, 100, regime=2)
+        wiring = r"{}/tables/RCPTT_{}128.dat".format(os.environ["ICPCONFIGROOT"], "wiring")
+
+        g.change_tables(wiring=wiring)
+        self.assertEqual(g.get_wiring_table(), wiring)
+
+    def test_GIVEN_valid_detector_table_to_change_tables_THEN_get_detector_table_returns_correct_file_path(self):
+
+        set_wait_for_complete_callback_dae_settings(True)
+        g.change_tcb(0, 10000, 100, regime=2)
+        detector = r"{}/tables/RCPTT_{}128.dat".format(os.environ["ICPCONFIGROOT"], "detector")
+        g.change_tables(detector=detector)
+        self.assertEqual(g.get_detector_table(), detector)
```

Comment:
The tables that set in the three tests above are the same that are already set in the `setUp` method. Could you please set different tables in the tests? This will make the assertion more meaningful.

Reply:
Have changed these in the latest commit, because the tables all need to be valid together changing just one individually is not a simple change. I have compiled the tests into one test which tests that all three file paths are correct.